clang/lib/Sema/SemaDecl.cpp

```diff
@@ -12236,6 +12236,23 @@ bool Sema::CheckFunctionDeclaration(Scope *S, FunctionDecl *NewFD,
     bool UsesSM = NewFD->hasAttr<ArmLocallyStreamingAttr>();
     bool UsesZA = Attr && Attr->isNewZA();
     bool UsesZT0 = Attr && Attr->isNewZT0();
+
+    if (UsesSM) {
+      if (NewFD->getReturnType()->isSizelessVectorType())
+        Diag(NewFD->getLocation(),
+             diag::warn_sme_locally_streaming_has_vl_args_returns);
+      auto *FPT = NewFD->getType()->castAs<FunctionProtoType>();
```

Comment:
Is `FunctionPrototype` correct here? That means the check does not happen on unprototyped functions, such as `__arm_locally_streaming void foo() { ... }` in C (not C++).

Reply:
Done.